ENH raise error for invalid 'install_cmd' (#714)
benchopt/benchopt: doc/benchmark_workflow/manage_dependencies.rst

.. _manage_dependencies:

> Manage dependencies

Reviewer: maybe call this "Install a benchmark"?
We will add later some info on how to install the benchmark and on conda env management

======================

In order to make it easy to run a new benchmark, benchopt provides an interface

> to easily specify and install requirements for the various component of the
> benchmarks.

Reviewer: ```suggestion
to specify and install requirements for the various components of the benchmark.
```

- The minimal requirements that are necessary to run the benchmark are

>   specified in ``objective.py``. They can be install using the command

Reviewer: ```suggestion
  specified in ``objective.py``. They can be installed using the command
```

  ``benchopt install --minimal``.

- The requirements that are specific to each ``Dataset/Solver`` can be
  specified in each class, and they can be installed individually by selecting
  the proper component using ``benchopt install -d dataset1 -s solver1``.

.. _specify_requirements:

Specify requirements
-----------------------

To specify how dependencies should be installed, you can use the ``install_cmd`` class attribute.
This attribute accepts two possible values:

1. ``conda`` (default): Dependencies will be installed using Conda. In this case, you should 
  specify the required dependencies in the ``requirements`` class attribute.

2. ``shell``: This option allows you to provide a custom shell script for installing dependencies. 
  When using this value, you need to set the ``install_script`` class attribute to the path of your shell script.
  Benchopt will execute this script in the shell and pass the Conda environment directory as an argument.

> Additionally, to define the dependencies required by your benchmark, you can use the ``requirements`` class attribute. 
> This attribute should list all necessary dependencies to ensure smooth execution and reproducibility of your benchmarks.

Reviewer: This is what is dscribed in the point 1. no?

```suggestion
```

By properly setting these attributes, you ensure that all dependencies are installed 

> correctly. This will help users to run your benchmarks without any issues.\n

Reviewer: ```suggestion
correctly. This will help users to run your benchmarks without any issues.
```
